packages/core/src/RepoSync.ts

```diff
@@ -202,255 +163,96 @@ export class RepoSync {
     }
   }
 
-  // Private methods
-
-  /**
-   * Called for each document upon initialization. Records the document's clock and advertises it.
-   * @param documentId
-   */
-  private async registerDoc(documentId: string) {
-    this.log('registerDoc', documentId)
-    const clock = await this.getClockFromDoc(documentId)
-
-    // Make sure we can sync this document
-    this.validateDoc(documentId, clock)
-
-    // Record the doc's initial clock
-    await this.updateClock(documentId, ours, clock)
-  }
-
-  /**
-   * Checks the local doc's clock to ensure that it can be synced
-   * @param documentId
-   * @param clock
-   */
-  private validateDoc(documentId: string, clock: Clock) {
-    this.log('validateDoc', documentId)
-
-    // Make sure doc has a clock (i.e. is an Automerge object)
-    if (!clock) throw new TypeError(ERR_NOCLOCK)
-
-    // Make sure the document is newer than what we already have
-    const ourClock = this.getOurClock(documentId)
-    if (isMoreRecent(ourClock, clock)) throw new RangeError(ERR_OLDCLOCK)
-  }
+  // PRIVATE METHODS
 
-  /**
-   * Event listener that fires when any document is modified on the repo
-   * @param documentId
-   */
+  /** Event listener that fires when any document is modified on the repo */
   private async onDocChanged(documentId: string) {
-    this.log('onDocChanged', documentId)
-    const clock = await this.getClockFromDoc(documentId)
-
-    // make sure we can sync the new document
-    this.validateDoc(documentId, clock)
-
     // send the document if peer doesn't have it or has an older version
     await this.maybeSendChanges(documentId)
-
-    // see if peer has a newer version
-    await this.maybeRequestChanges(documentId, clock)
-
```

Comment:
`maybeRequestChanges` was not really doing what it said it was and wasn't needed. If we need changes we know it and we just ask for them.